pkg/gcp/validation/validation.go

```diff
@@ -112,6 +114,16 @@ func validateGCPDisks(disks []*api.GCPDisk, fldPath *field.Path) []error {
 		if disk.Boot && "" == disk.Image {
 			allErrs = append(allErrs, field.Required(idxPath.Child("image"), "image is required for boot disk"))
 		}
+		if disk.Encryption != nil {
+			var kmsKeyName = strings.TrimSpace(disk.Encryption.KmsKeyName)
+			var kmsKeyServiceAccount = strings.TrimSpace(disk.Encryption.KmsKeyServiceAccount)
+			if kmsKeyName == "" {
+				allErrs = append(allErrs, field.Required(idxPath.Child("kmsKeyName"), "kmsKeyName is required to be specified"))
+			}
+			if disk.Encryption.KmsKeyServiceAccount != "" && kmsKeyServiceAccount == "" {
+				allErrs = append(allErrs, field.Required(idxPath.Child("kmsKeyServiceAccount"), "kmsKeyServiceAccount should either be explicitly specified or left un-specified to default to the Compute Service Agent"))
+			}
```

Comment:
this is the case where customer provides spaces , right?
```suggestion
// to deal with situation where  just spaces have been specified for `kmsKeyServiceAccount`
			if disk.Encryption.KmsKeyServiceAccount != "" && kmsKeyServiceAccount == "" {
				allErrs = append(allErrs, field.Required(idxPath.Child("kmsKeyServiceAccount"), "kmsKeyServiceAccount should either be explicitly specified without spaces or left un-specified to default to the Compute Service Agent"))
			}
```